### PR TITLE
Make use of github-actions

### PR DESCRIPTION
**Pull request by BatuAksoy**
- Organised docker files under `Docker` folder. Modified compose files to work on appropriate contexts.
- Made use of test specific docker components in test workflow (I have seen that at some point in past we switched a without-docker setup, just wanted to try again if duration is acceptable - let me know if you think I have to switch back to without-docker setup)
- Removed travis related files
- Added a release action. (When a release is published, binary file is populated from workflow)

### CODE REVIEW

**Comment by cenkalti** [comment]
`make test` does not work. Can you fix it?
```
$ make test
...
docker-compose -f ./Docker/docker-compose-test.yml run --rm test
[+] Running 3/3
 ⠿ Container docker-rabbitmq-1  Created                                                                                                                                    0.7s
 ⠿ Container docker-mysql-1     Created                                                                                                                                    1.0s
 ⠿ Container docker-tracker-1   Created                                                                                                                                    0.5s
[+] Running 3/3
 ⠿ Container docker-mysql-1     Started                                                                                                                                    0.7s
 ⠿ Container docker-rabbitmq-1  Started                                                                                                                                    0.7s
 ⠿ Container docker-tracker-1   Started                                                                                                                                    0.7s
MySQL is ready.
panic: read /etc/efes.toml: is a directory

goroutine 1 [running]:
github.com/putdotio/efes.init.1()
	/go/src/github.com/putdotio/efes/config_test.go:14 +0x1e5
FAIL	github.com/putdotio/efes	0.104s
FAIL
make: *** [test] Error 1
```

I think there is a problem with Docker volume mapping.

File: Makefile [hunk 1]
```diff
@@ -5,14 +5,23 @@ NAME := efes
...
-	docker build -t $(NAME) -f Dockerfile .
-	docker-compose rm -fsv
-	docker-compose up --build
+	docker build -t efes-base -f ./Docker/efes-base/Dockerfile .
+	docker-compose -f ./Docker/docker-compose.yml rm -fsv
+	docker-compose -f ./Docker/docker-compose.yml up --build
 
 test:
-	docker-compose rm -fsv
-	docker-compose -f docker-compose-test.yml up --build --exit-code-from test --abort-on-container-exit
```
Comment (cenkalti) [inline Makefile:15]: Why did you remove these parameters?

**Comment by BatuAksoy** [comment]
Weird, works for me. I will check again.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    types:
+      - published
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build and copy binary
+        run: |
```
Comment (cenkalti) [inline .github/workflows/release.yaml]: Is it possible to utilize `make build-docker` command here?

File: Docker/test/Dockerfile [hunk 3]
```diff
@@ -5,7 +5,6 @@ ENV GO111MODULE=on
 COPY go.mod .
 COPY go.sum .
 RUN go mod download
-ADD docker-run-tests.sh /root/run-tests.sh
+ADD ./Docker/test/docker-run-tests.sh /root/run-tests.sh
 COPY . .
-ADD config-docker.toml /etc/efes.toml
```
Comment (cenkalti) [inline Docker/test/Dockerfile:10]: Is it okay to run tests without a config?

File: Makefile (hunk 1 above)
Comment (BatuAksoy) [reply Makefile:15]: I have split this command into two, `docker-compose build` and `docker-compose run test`
Since it runs test container directly, exit-code comes directly from container now.

But I realized that, containers do not stop when the test container stops. So `--abort-on-container-exit` behavior is missing. I will add a command to remove containers afterwards.

File: Docker/test/Dockerfile (hunk 3 above)
Comment (BatuAksoy) [reply Docker/test/Dockerfile:10]: Tests will fail in that case. The reason why I removed this line is because we share config for `efes-tracker` and `test` container in tests. (and additionally `efes-server`)
I wanted all those to share config, only solution made sense for me was putting the config into the container through a volume. 

I am open to anything else, if you have any other ideas please share.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (BatuAksoy) [reply .github/workflows/release.yaml]: I tried to utilize it first and tried hard. But variable set in the `build-docker` command (`CONTAINER_ID`) couldn't be utilized in following lines.

I have tried this command on both linux and macOS, they worked but weird it does not work on CI. CI output here: https://github.com/BatuAksoy/efes/runs/8171343849?check_suite_focus=true#step:3:83

File: Docker/docker-compose-test.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  rabbitmq:
+    build:
+      context: ./rabbitmq
+    environment:
+      RABBITMQ_DEFAULT_USER: efes
+      RABBITMQ_DEFAULT_PASS: 123
+      RABBITMQ_NODENAME: efes@localhost
+    logging:
+      driver: none
+  tracker:
```
Comment (cenkalti) [inline Docker/docker-compose-test.yml]: There was no tracker service in old setup. Why do we require it now?

File: .github/workflows/release.yaml (hunk 2 above)
Comment (BatuAksoy) [reply .github/workflows/release.yaml]: Let me try again

File: Docker/test/Dockerfile (hunk 3 above)
Comment (cenkalti) [reply Docker/test/Dockerfile:10]: I'd rather put config into images rather that using volume mount.

File: Docker/docker-compose-test.yml (hunk 4 above)
Comment (BatuAksoy) [reply Docker/docker-compose-test.yml]: Good catch! I blindly replaced all localhosts with container names. Where I shouldn't have done for tracker url in config. Will fix it.

File: Makefile (hunk 1 above)
Comment (BatuAksoy) [reply Makefile:15]: I switched back to `up` command again since it was more clean.

File: Docker/test/Dockerfile (hunk 3 above)
Comment (BatuAksoy) [reply Docker/test/Dockerfile:10]: We now have config files for each image folder and they are included in the image.

**Comment by BatuAksoy** [comment]
@cenkalti One missing thing is goveralls setup. I will work on that after this PR, if it's ok
